pkg/platform/runtime/setup/setup.go

```diff
@@ -230,14 +240,35 @@ func (s *Setup) Update() (rerr error) {
 	return nil
 }
 
-func (s *Setup) updateArtifacts() ([]artifact.ArtifactID, error) {
+// Panics are serious, and reproducing them in the runtime package is HARD. To help with this we dump
+// the build plan when a panic occurs so we have something more to go on.
+func (s *Setup) solveUpdateRecover(r interface{}) {
+	if r == nil {
+		return
+	}
+	buildplan, err := s.store.BuildPlanRaw()
```

Comment:
Right now, we are storing the build plan fairly late in the setup process (see [here](https://github.com/ActiveState/cli/blob/master/pkg/platform/runtime/setup/setup.go#L620)). If it's easy we should consider moving this up to happen earlier or file a follow up to do this in a different way. Regardless there is higher potential that we won't have the raw build plan from the `store` with the current implementation.

Reply:
Are you suggesting this as a general improvement, or you see a potential way for this PR to trip over this mechanic? If the former; that's a good idea - should file it on Jira! If the latter; can you elaborate on the issue you foresee?

Reply:
Definitely a general improvement. Right now, depending on where the setup process failed we may not have the raw build plan because we are relying on the store. We could store it earlier in the process or make it part of the build response so we're almost guaranteed to have it. I'll file a story in Jira.